src/components/app/index.tsx

```diff
@@ -2,17 +2,58 @@ import { h, Component } from 'preact';
 import { bind, bitmapToImageData } from '../../lib/util';
 import * as style from './style.scss';
 import Output from '../output';
+import Options from '../options';
 
-import {MozJpegEncoder} from '../../lib/codec-wrappers/mozjpeg-enc';
+// import { Encoder } from '../../lib/codec-wrappers/codec';
+import { MozJpegEncoder } from '../../lib/codec-wrappers/mozjpeg-enc';
+
+export type ImageType = 'original' | 'jpeg';
+
+export type CodecOptions = any;
+
+type Encoders = {
+  [type: string]: any  // @todo Encoder
```

Comment:
This is a "TypeScript is frustrating" moment. For some reason, MozJpegEncoder isn't considered to have the same interface as Encoder?  It's likely just me doing something stupid.

Reply:
`MozJpegEncoder` doesn’t implement `Encoder`, an instance of `MozJpegEncoder` does 😄 

You could do `new () => Encoder` instead of any here.